test/max_tick_cross.test.ts

```diff
@@ -0,0 +1,648 @@
+        i,
+        i + 10n,
+        liquidityDelta,
+        slippageLimit,
+        slippageLimit
+      )
+    }
+
+    await withdrawTokens(swapper, [tokenX, amount])
+
+    const { targetSqrtPrice } = await quote(invariant, poolKey, xToY, amount, byAmountIn, slippage)
+
+    const poolBefore = await getPool(invariant, poolKey)
+    const { gasAmount } = await initSwap(
+      invariant,
+      swapper,
+      poolKey,
+      xToY,
+      amount,
+      byAmountIn,
+      targetSqrtPrice
+    )
+    const poolAfter = await getPool(invariant, poolKey)
+    const crosses = (poolAfter.currentTickIndex - poolBefore.currentTickIndex) / -10n
+    expect(crosses).toBe(13n)
+    expect(gasAmount).toBeGreaterThan(txGasLimit)
+  }, 100000)
+  test('max tick cross swap yToX and ByAmountIn, liquidity gap between positions', async () => {
+    const lastInitializedTick = 240n
+    const amount = 40000n
+    const xToY = false
+    const slippage = MaxSqrtPrice
+    const byAmountIn = true
+
+    for (let i = 0n; i < lastInitializedTick; i += 20n) {
+      const positionOwnerBalanceX = await balanceOf(tokenX.contractId, positionOwner.address)
+      const positionOwnerBalanceY = await balanceOf(tokenY.contractId, positionOwner.address)
+      const { sqrtPrice: slippageLimit } = await getPool(invariant, poolKey)
+      await initPosition(
+        invariant,
+        positionOwner,
+        poolKey,
+        positionOwnerBalanceX,
+        positionOwnerBalanceY,
+        i,
+        i + 10n,
+        liquidityDelta,
+        slippageLimit,
+        slippageLimit
+      )
+    }
+
+    await withdrawTokens(swapper, [tokenY, amount])
+
+    const { targetSqrtPrice } = await quote(invariant, poolKey, xToY, amount, byAmountIn, slippage)
+
+    const poolBefore = await getPool(invariant, poolKey)
+    const { gasAmount } = await initSwap(
+      invariant,
+      swapper,
+      poolKey,
+      xToY,
+      amount,
+      byAmountIn,
+      targetSqrtPrice
+    )
+
+    const poolAfter = await getPool(invariant, poolKey)
+    const crosses = (poolAfter.currentTickIndex - poolBefore.currentTickIndex) / 10n
+    expect(crosses).toBe(14n)
+    expect(gasAmount).toBeGreaterThan(txGasLimit)
+  }, 100000)
+  test('max tick cross swap xToY and ByAmountIn, positions between search limit range', async () => {
+    const lastInitializedTick = -35000n
+    const amount = 58100n
+    const xToY = true
+    const slippage = MinSqrtPrice
+    const byAmountIn = true
+
+    for (let i = lastInitializedTick; i < 0n; i += searchLimit) {
+      const positionOwnerBalanceX = await balanceOf(tokenX.contractId, positionOwner.address)
+      const positionOwnerBalanceY = await balanceOf(tokenY.contractId, positionOwner.address)
+      const { sqrtPrice: slippageLimit } = await getPool(invariant, poolKey)
+      await initPosition(
+        invariant,
+        positionOwner,
+        poolKey,
+        positionOwnerBalanceX,
+        positionOwnerBalanceY,
+        i,
```

Comment:
Why is this test case designed this way?
The positions are narrow but far apart from each other. IMO It is better to use wide positions without gaps.